src/ArcPay.sol

```diff
@@ -53,6 +56,7 @@ contract ArcPay is Ownable2Step {
         root = mintTree.insert({
             leaf: PoseidonT3.hash([uint(uint160(receiver)), msg.value])
         });
+        emit Mint(receiver, msg.value);
```

Comment:
Are we planning on keeping this in the actual code? I think it's honestly fine, since the mint cost is probably dominated by sending Eth, so this is negligible and makes our infra easier.

Reply:
yeah, planning to include in the code. It simplifies monitoring for mints by a lot.